test/functional/specs/desktop/C2582.js

```diff
@@ -0,0 +1,58 @@
+import { t, Selector, ClientFunction } from "testcafe";
+import createNetworkLogger from "../../helpers/networkLogger";
+import getResponseBody from "../../helpers/networkLogger/getResponseBody";
+import { responseStatus } from "../../helpers/assertions/index";
+import fixtureFactory from "../../helpers/fixtureFactory";
+import testServerUrl from "../../helpers/constants/testServerUrl";
+
+const urlCollector = `${testServerUrl}/test/functional/sandbox/html/alloySdk.html`;
+
+const networkLogger = createNetworkLogger();
+
+fixtureFactory({
+  title:
+    "C2582 - When ECID is available on client, allow all requests to be promptly sent using the ECID",
+  url: urlCollector,
+  requestHooks: [
+    networkLogger.edgeEndpointLogs,
+    networkLogger.gatewayEndpointLogs
+  ]
+});
+
+test.meta({
+  ID: "C2582",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+test("Test C2582: Once the identity cookie is available, requests should go out promptly (in parallel) and should contain the ECID.", async () => {
```

Comment:
The titles being passed into `fixtureFactory` and `test` (not in just this file but others) seem similar but not the same. What's our intended pattern/logic for choosing a title for each?

Reply:
The title seems to suggest to me that the test would be as follows:

1. User clicks the event button three times.
2. Only one request goes out.
3. The response contains the ECID and the identity cookie.
4. Request #2 and #3 go out simultaneously with the ECID.

while this seems to be what's actually being tested:

1. User clicks the event button once.
2. One request goes out.
3. The response contains the ECID and the identity cookie.
4. The user clicks the event button again.
5. The second request goes out with the ECID.

I would prefer we test the former rather than the latter.

Reply:
> The titles being passed into `fixtureFactory` and `test` (not in just this file but others) seem similar but not the same. What's our intended pattern/logic for choosing a title for each?

If I understood the benefit that test rail gives us, perhaps I could have a better opinion on this.  I think we will make more work for ourselves if we require only one test per file.  More work to make more test rail items, and more work to setup each test in the test files.